src/quick-lint-js/error.h

```diff
@@ -116,6 +116,13 @@
                 "C-style for loop is missing its third component"),            \
             expected_last_component))                                          \
                                                                                \
+  QLJS_ERROR_TYPE(                                                             \
+      error_cannot_assign_to_loop_variable_in_for_of_or_in_loop, "E173",       \
+      { source_code_span equal_token; },                                       \
+      ERROR(QLJS_TRANSLATABLE(                                                 \
+                "cannot assign to loop variable in for of/in loop"),           \
```

Comment:
Minor: Maybe we should make a `statement_kind` for improve error messages. We can do this in a separate PR.